tests/Temporalio.Tests/Testing/WorkflowEnvironmentTests.cs

```diff
@@ -0,0 +1,217 @@
+#pragma warning disable SA1201, SA1204 // We want to have classes near their tests
+
+namespace Temporalio.Tests.Testing;
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Temporalio.Activities;
+using Temporalio.Api.Enums.V1;
+using Temporalio.Client;
+using Temporalio.Exceptions;
+using Temporalio.Testing;
+using Temporalio.Worker;
+using Temporalio.Workflows;
+using Xunit;
+using Xunit.Abstractions;
+
+public class WorkflowEnvironmentTests : TestBase
+{
+    // Time-skipping test server only runs on x86/x64
+    public sealed class OnlyIntelFactAttribute : FactAttribute
```

Comment:
nit: it's not only intel, amd has this arch too but 🤷

Reply:
"Intel" is a common term for the instruction set, even though it's AMD's 64-bit extensions. I just used the word they are using to describe it at https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.architecture. This is test only code anyways. I could changed to `OnlyIntelBasedFactAttrbiute` or something.